control-plane-operator/controllers/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -389,6 +389,19 @@ func reconcileAWSEndpointService(ctx context.Context, awsEndpointService *hyperv
 			}
 			return err
 		}
+
+		if aws.StringValue(output.VpcEndpoints[0].ServiceName) != awsEndpointService.Status.EndpointServiceName {
+			log.Info("endpoint links to wrong endpointservice, deleting...", "VPCEndpointServiceName", aws.StringValue(output.VpcEndpoints[0].ServiceName), "AWSEndpointServiceName", awsEndpointService.Status.EndpointServiceName)
```

Comment:
nit: using `VPCEndpointServiceName` and `AWSEndpointServiceName` as keys seems confusing to me here, as a reader of the logs I can't tell what represents what.
I think it might be more clear something like `LinkedVPCEndpointServiceName` and `WantedVPCEndpointService`